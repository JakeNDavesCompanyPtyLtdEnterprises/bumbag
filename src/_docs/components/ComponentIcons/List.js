import React from 'react';

const SvgComponent = props => (
  <svg width={63} height={51} fill="none" {...props}>
    <path
      d="M63 5H13v8h50V5zM.516 11.236h1.008V7.24l-1.008.072v-1.56c.352 0 .732-.064 1.14-.192a4.976 4.976 0 0 0 1.188-.588h1.26v6.264h1.008V13H.516v-1.764zm6.686 1.836c-.424 0-.756-.116-.996-.348a1.167 1.167 0 0 1-.36-.876c0-.36.124-.652.372-.876.248-.224.576-.336.984-.336.4 0 .724.12.972.36.248.232.372.516.372.852 0 .376-.116.676-.348.9-.232.216-.564.324-.996.324zM63 22H13v8h50v-8zM.456 28.308c.944-.848 1.656-1.568 2.136-2.16.48-.592.72-1.084.72-1.476 0-.256-.08-.452-.24-.588-.152-.136-.36-.204-.624-.204-.496 0-1.012.156-1.548.468L.312 22.56a6.55 6.55 0 0 1 1.296-.504 5.382 5.382 0 0 1 1.356-.168c.56 0 1.048.112 1.464.336.424.216.748.516.972.9.224.376.336.808.336 1.296 0 .728-.244 1.408-.732 2.04-.488.624-1.224 1.22-2.208 1.788h2.892V30H.456v-1.692zm7.53 1.764c-.423 0-.755-.116-.995-.348a1.167 1.167 0 0 1-.36-.876c0-.36.124-.652.372-.876.248-.224.576-.336.984-.336.4 0 .724.12.972.36.248.232.372.516.372.852 0 .376-.116.676-.348.9-.232.216-.564.324-.996.324zM63 39H13v8h50v-8zM2.772 47.132a6.117 6.117 0 0 1-2.412-.468L.756 44.9c.632.256 1.22.384 1.764.384.656 0 .984-.244.984-.732 0-.344-.168-.588-.504-.732-.336-.144-.896-.216-1.68-.216v-1.488c.712 0 1.244-.072 1.596-.216.36-.144.54-.348.54-.612 0-.36-.308-.54-.924-.54-.584 0-1.188.136-1.812.408l-.468-1.632a6.212 6.212 0 0 1 2.772-.648c.736 0 1.332.176 1.788.528.464.352.696.832.696 1.44 0 .392-.104.748-.312 1.068-.2.32-.472.576-.816.768a2.69 2.69 0 0 1 1.08.852c.272.36.408.76.408 1.2 0 .488-.132.912-.396 1.272-.256.36-.62.64-1.092.84-.464.192-1 .288-1.608.288zm5.285-.06c-.424 0-.756-.116-.996-.348a1.167 1.167 0 0 1-.36-.876c0-.36.124-.652.372-.876.248-.224.576-.336.984-.336.4 0 .724.12.972.36.248.232.372.516.372.852 0 .376-.116.676-.348.9-.232.216-.564.324-.996.324z"
      fill="#3926A5"
    />
  </svg>
);

export default SvgComponent;
