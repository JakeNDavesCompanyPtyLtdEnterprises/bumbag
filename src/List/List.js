// @flow
import React, { type Node } from 'react';

import type { Palette } from '../typesold';
import _List from './styled';
import ListItem from './ListItem';

type Props = {
  children: Node,
  className?: string,
  color?: Palette,
  isOrdered?: boolean,
  isHorizontal?: boolean
};

const List = ({ children, className, isOrdered, ...props }: Props) => (
  <_List use={isOrdered ? 'ol' : undefined} className={className} isOrdered={isOrdered} {...props}>
    {children}
  </_List>
);

List.Item = ListItem;

List.defaultProps = {
  className: undefined,
  color: undefined,
  isOrdered: false,
  isHorizontal: false
};

export default List;
