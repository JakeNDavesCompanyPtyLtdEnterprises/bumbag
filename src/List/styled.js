import styled, { css } from 'reakit/styled';
import Box from 'reakit/Box';
import List from 'reakit/List';
import { theme } from 'styled-tools';

const orderedProperties = css`
  list-style-type: decimal;

  & & {
    list-style-type: lower-alpha;
  }

  & & & {
    list-style-type: lower-roman;
  }

  & {
    ${theme('fannypack.List.ordered')};
  }
`;

const horizontalProperties = css`
  & li {
    display: inline-block;
    margin-right: 1rem;
    margin-bottom: unset;
  }

  & {
    ${theme('fannypack.List.horizontal')};
  }
`;

export const ListItem = styled(Box)`
  margin-bottom: 0.2rem;

  & {
    ${theme('fannypack.List.item.base')};
  }
`;

export default styled(List)`
  list-style: unset;
  list-style-type: none;

  & & {
    margin-left: 1rem;
  }

  & {
    ${props => props.isOrdered && orderedProperties};
  }

  & {
    ${props => props.isHorizontal && horizontalProperties};
  }

  & {
    ${theme('fannypack.List.base')};
  }
`;
