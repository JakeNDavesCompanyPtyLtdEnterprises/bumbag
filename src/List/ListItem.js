// @flow
import React, { type Node } from 'react';

import type { Palette } from '../typesold';
import { ListItem as _ListItem } from './styled';

type Props = {
  children: Node,
  className?: string,
  color?: Palette
};

const ListItem = ({ children, className, ...props }: Props) => (
  <_ListItem use="li" className={className} {...props}>
    {children}
  </_ListItem>
);

ListItem.defaultProps = {
  className: undefined,
  color: undefined
};

export default ListItem;
