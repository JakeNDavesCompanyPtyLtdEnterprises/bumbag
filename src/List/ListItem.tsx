import * as React from 'react';
// @ts-ignore
import PropTypes from 'prop-types';
import { BoxProps as ReakitBoxProps } from 'reakit/ts';

import { ListItem as _ListItem } from './styled';

export interface LocalListItemProps {
  children: React.ReactNode;
  className?: string;
}
export type ListItemProps = ReakitBoxProps & LocalListItemProps;

export const ListItem: React.FunctionComponent<LocalListItemProps> = ({ children, className, ...props }) => (
  <_ListItem use="li" className={className} {...props}>
    {children}
  </_ListItem>
);

ListItem.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
};
ListItem.defaultProps = {
  className: undefined
};

const C: React.FunctionComponent<ListItemProps> = ListItem;
export default C;
