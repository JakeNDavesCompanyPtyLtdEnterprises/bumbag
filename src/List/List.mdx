---
name: List
route: /typography/list
menu: Typography
---

import { Playground, PropsTable } from 'docz';
import { Box } from '../primitives/index';
import {List} from './List';
import {ListItem} from './ListItem';
import Paragraph from '../Paragraph';

# List

## Import

`import { List } from 'fannypack'`

## Basic Usage

`<List>` renders a `<ul>` element by default with reset styles and `list-style` unset.

<Playground>
  <Paragraph>My awesome list:</Paragraph>
  <List>
    <List.Item>Item 1</List.Item>
    <List.Item>Item 2</List.Item>
    <List.Item>Item 3</List.Item>
    <List.Item>Item 4</List.Item>
  </List>
</Playground>

## Types

Specify if you want your list items to have a style type with the `listStyleType` CSS property.

[Click here](https://developer.mozilla.org/en-US/docs/Web/CSS/list-style-type) to see all the `listStyleType` values.

<Playground>
  <Paragraph>My awesome list:</Paragraph>
  <List listStyleType="disc" listStylePosition="inside">
    <List.Item>Item 1</List.Item>
    <List.Item>Item 2</List.Item>
  </List>
  <List listStyleType="circle" listStylePosition="inside">
    <List.Item>Item 1</List.Item>
    <List.Item listStyleType="georgian">Item 2</List.Item>
    <List listStyleType="circle">
      <List.Item>Item 1</List.Item>
      <List.Item>Item 2</List.Item>
    </List>
  </List>
</Playground>

## Ordered lists

To use an ordered list, use the `isOrdered` prop.

<Playground>
  <Paragraph>My awesome list:</Paragraph>
  <List isOrdered listStylePosition="inside">
    <List.Item>Item 1</List.Item>
    <List.Item>Item 2</List.Item>
    <List isOrdered>
      <List.Item>Item 3</List.Item>
      <List isOrdered>
        <List.Item>Item 4</List.Item>
      </List>
    </List>
  </List>
</Playground>

## Horizontal lists

To use an horizontal list, use the `isHorizontal` prop.

<Playground>
  <List isHorizontal>
    <List.Item>Item 1</List.Item>
    <List.Item>Item 2</List.Item>
  </List>
</Playground>


## `<List>` Props

<PropsTable of={List} />

## `<List.Item>` Props

<PropsTable of={ListItem} />

## Theming

### Schema

```jsx
{
  base: string | Object,
  item: {
    base: string | Object
  },
  ordered: string | Object
}
```
