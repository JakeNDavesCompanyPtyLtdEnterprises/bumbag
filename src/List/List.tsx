import * as React from 'react';
import { ListProps as ReakitListProps } from 'reakit/ts';

import _List from './styled';
import ListItem, { ListItemProps } from './ListItem';

export interface LocalListProps {
  children: React.ReactNode;
  className?: string;
  isOrdered?: boolean;
  isHorizontal?: boolean;
}
export type ListProps = LocalListProps & ReakitListProps;
export interface ListComponents {
  Item: React.FunctionComponent<ListItemProps>;
}

export const List: React.FunctionComponent<LocalListProps> & ListComponents = ({
  children,
  className,
  isOrdered,
  ...props
}) => (
  <_List use={isOrdered ? 'ol' : undefined} className={className} isOrdered={isOrdered} {...props}>
    {children}
  </_List>
);

List.Item = ListItem;

List.defaultProps = {
  className: undefined,
  isOrdered: false,
  isHorizontal: false
};

const C: React.FunctionComponent<ListProps> & ListComponents = List;
export default C;
