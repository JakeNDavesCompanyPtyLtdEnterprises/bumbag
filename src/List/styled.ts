import { List } from 'reakit';
import { theme } from 'styled-tools';

import { Box } from '../primitives';
import { styled, css } from '../styled';
import { ListItemProps } from './ListItem';
import { ListProps } from './List';

const orderedProperties = css`
  list-style-type: decimal;

  & & {
    list-style-type: lower-alpha;
  }

  & & & {
    list-style-type: lower-roman;
  }

  & {
    ${theme('fannypack.List.ordered')};
  }
`;

const horizontalProperties = css`
  & li {
    display: inline-block;
    margin-right: 1rem;
    margin-bottom: unset;
  }

  & {
    ${theme('fannypack.List.horizontal')};
  }
`;

export const ListItem = styled(Box)<ListItemProps>`
  margin-bottom: 0.2rem;

  & {
    ${theme('fannypack.List.item.base')};
  }
`;

export default styled(List)<ListProps>`
  list-style: unset;
  list-style-type: none;

  & & {
    margin-left: 1rem;
  }

  & {
    ${props => props.isOrdered && orderedProperties};
  }

  & {
    ${props => props.isHorizontal && horizontalProperties};
  }

  & {
    ${theme('fannypack.List.base')};
  }
`;
