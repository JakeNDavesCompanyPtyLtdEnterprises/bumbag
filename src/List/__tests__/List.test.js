import React from 'react';
import render from '../../_utils/tests/render';
import List from '../List';
import 'jest-styled-components';

it('renders correctly for a basic list', () => {
  const { container } = render(
    <List>
      <List.Item>Item 1</List.Item>
      <List.Item>Item 2</List.Item>
      <List.Item>Item 3</List.Item>
      <List.Item>Item 4</List.Item>
    </List>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a list with styles', () => {
  const { container } = render(
    <div>
      <List listStyleType="disc" listStylePosition="inside">
        <List.Item>Item 1</List.Item>
        <List.Item>Item 2</List.Item>
      </List>
      <List listStyleType="circle" listStylePosition="inside">
        <List.Item>Item 1</List.Item>
        <List.Item listStyleType="georgian">Item 2</List.Item>
        <List listStyleType="circle">
          <List.Item>Item 1</List.Item>
          <List.Item>Item 2</List.Item>
        </List>
      </List>
    </div>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for ordered lists', () => {
  const { container } = render(
    <div>
      <List isOrdered listStylePosition="inside">
        <List.Item>Item 1</List.Item>
        <List.Item>Item 2</List.Item>
        <List isOrdered>
          <List.Item>Item 3</List.Item>
          <List isOrdered>
            <List.Item>Item 4</List.Item>
          </List>
        </List>
      </List>
    </div>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for horizontal lists', () => {
  const { container } = render(
    <div>
      <List isHorizontal>
        <List.Item>Item 1</List.Item>
        <List.Item>Item 2</List.Item>
        <List.Item>Item 3</List.Item>
        <List.Item>Item 4</List.Item>
      </List>
    </div>
  );
  expect(container.firstChild).toMatchSnapshot();
});
