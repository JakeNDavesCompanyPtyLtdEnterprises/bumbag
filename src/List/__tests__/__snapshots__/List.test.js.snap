// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for a basic list 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #586ccf;
  outline-offset: 2px;
}

.c3 {
  margin-bottom: 0.2rem;
}

.c2 {
  list-style: unset;
  list-style-type: none;
}

.c2 .c2 {
  margin-left: 1rem;
}

<div
  class="c0 c1"
>
  <ul
    class="c2 c1"
  >
    <li
      class="c3 c1"
    >
      Item 1
    </li>
    <li
      class="c3 c1"
    >
      Item 2
    </li>
    <li
      class="c3 c1"
    >
      Item 3
    </li>
    <li
      class="c3 c1"
    >
      Item 4
    </li>
  </ul>
</div>
`;

exports[`renders correctly for a list with styles 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #586ccf;
  outline-offset: 2px;
}

.c3 {
  margin-bottom: 0.2rem;
}

.c2 {
  list-style: unset;
  list-style-type: none;
}

.c2 .c2 {
  margin-left: 1rem;
}

<div
  class="c0 c1"
>
  <div>
    <ul
      class="c2 c1"
      style="list-style-type: disc; list-style-position: inside;"
    >
      <li
        class="c3 c1"
      >
        Item 1
      </li>
      <li
        class="c3 c1"
      >
        Item 2
      </li>
    </ul>
    <ul
      class="c2 c1"
      style="list-style-type: circle; list-style-position: inside;"
    >
      <li
        class="c3 c1"
      >
        Item 1
      </li>
      <li
        class="c3 c1"
        style="list-style-type: georgian;"
      >
        Item 2
      </li>
      <ul
        class="c2 c1"
        style="list-style-type: circle;"
      >
        <li
          class="c3 c1"
        >
          Item 1
        </li>
        <li
          class="c3 c1"
        >
          Item 2
        </li>
      </ul>
    </ul>
  </div>
</div>
`;

exports[`renders correctly for horizontal lists 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #586ccf;
  outline-offset: 2px;
}

.c3 {
  margin-bottom: 0.2rem;
}

.c2 {
  list-style: unset;
  list-style-type: none;
}

.c2 .c2 {
  margin-left: 1rem;
}

.c2 li {
  display: inline-block;
  margin-right: 1rem;
  margin-bottom: unset;
}

<div
  class="c0 c1"
>
  <div>
    <ul
      class="c2 c1"
    >
      <li
        class="c3 c1"
      >
        Item 1
      </li>
      <li
        class="c3 c1"
      >
        Item 2
      </li>
      <li
        class="c3 c1"
      >
        Item 3
      </li>
      <li
        class="c3 c1"
      >
        Item 4
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders correctly for ordered lists 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #586ccf;
  outline-offset: 2px;
}

.c3 {
  margin-bottom: 0.2rem;
}

.c2 {
  list-style: unset;
  list-style-type: none;
}

.c2 .c2 {
  margin-left: 1rem;
}

.c2 {
  list-style-type: decimal;
}

.c2 .c2 {
  list-style-type: lower-alpha;
}

.c2 .c2 .c2 {
  list-style-type: lower-roman;
}

<div
  class="c0 c1"
>
  <div>
    <ol
      class="c2 c1"
      style="list-style-position: inside;"
    >
      <li
        class="c3 c1"
      >
        Item 1
      </li>
      <li
        class="c3 c1"
      >
        Item 2
      </li>
      <ol
        class="c2 c1"
      >
        <li
          class="c3 c1"
        >
          Item 3
        </li>
        <ol
          class="c2 c1"
        >
          <li
            class="c3 c1"
          >
            Item 4
          </li>
        </ol>
      </ol>
    </ol>
  </div>
</div>
`;
